tests/rest/client/test_room_access_rules.py

```diff
@@ -320,14 +309,33 @@ def test_restricted(self):
             expected_code=200,
         )
 
+        # We are allowed to publish the room to the public room list
+        url = "/_matrix/client/r0/directory/list/room/%s" % self.restricted_room
+        data = {"visibility": "public"}
+
+        request, channel = self.make_request("PUT", url, data, access_token=self.tok)
+        self.render(request)
+        self.assertEqual(channel.code, 200, channel.result)
+
+        # We are allowed to remove the room from the public room list
+        url = "/_matrix/client/r0/directory/list/room/%s" % self.restricted_room
+        data = {"visibility": "private"}
+
+        request, channel = self.make_request("PUT", url, data, access_token=self.tok)
+        self.render(request)
+        self.assertEqual(channel.code, 200, channel.result)
+
     def test_direct(self):
         """Tests that, in direct mode, other users than the initial two can't be invited,
         but the following scenario works:
           * invited user joins the room
           * invited user leaves the room
           * room creator re-invites invited user
-        Also tests that a user from a HS that's in the list of forbidden domains (to use
+
+        Tests that a user from a HS that's in the list of forbidden domains (to use
         in restricted mode) can be invited.
+
+        Tests that the room cannot be published to the public room list.
```

Comment:
Also needs a test to make sure we can't change the join rules to something other than `invite`.

Reply:
Turns out there was a test for this that I had removed. Re-added it back!